### ISSUE TITLE
accuracy benchmarks for deep models

### ISSUE THREAD

**Issue by daviddao** [tflearn/tflearn#126]
@fdiehl and I ran the `residual-network-cifar10.py` example but realized that it doesn't converge to the expected 94% in literature. It actually get stuck with 83%.

![screen shot 2016-06-02 at 23 43 35](https://cloud.githubusercontent.com/assets/1241240/15776227/9f45f2cc-2986-11e6-9b3a-d7a285788ef7.png)

@aymericdamien do you have any ideas?

In order for more people to use that great library, we should also have benchmarks for the examples we implemented (and hopefully guarantee that they work). Nervana Neon does this in a nice way with its examples (as a code comment) and the results are reproducible.

**Comment by aymericdamien** [comment]
I agree, results should match literature. 
First, I think there is an issue with the learning rate which doesn't reduce by itself, indeed if you stop training and reset learning rate and restart training, then you get better performances.
Also we should delete the old preprocessing and use the new `DataPreprocessing` and `DataAugmentation`, that should also give better performances. At last, I think that TensorFlow fixed the issue where strides could not be larger than filter_size, so we could remove the average pooling.
I will try to reimplement it to get better results.

**Comment by aymericdamien** [comment]
@daviddao There is indeed an issue with learning rate decay. If I train for 100 epochs, and set lr to 0.01, load model and retrain, I can get similar result with resnet paper. 
Like the paper, I zero center data and use augmentation (random_rotation instead of random_crop as it seems to have some issues that I am investigating now...)
I will update the example when that is fix!

**Comment by daviddao** [comment]
That's strange! But glad to hear that you manage to achieve similar results :).

**Comment by aymericdamien** [comment]
@daviddao I updated resnet code, and fix lr decay issue. Now I can get similar results with the original paper. I modified resnet to use padding and avg pooling, but it would be nice to try the projection too. You can change 'n=5' parameter in the example to set the desired amount of layers you want.
As you suggested, we can add the benchmark to the script (or maybe make a special section for that).

**Comment by daviddao** [comment]
Thanks, I am eager to try it out!

**Comment by daviddao** [comment]
Update: @aymericdamien I could achieve 92 % accuracy with 32 layers, which is similar to the literature.

However, when I ran 110 layers (n = 9), I also get stuck at 92% starting from epoch 70:

![screen shot 2016-06-10 at 11 09 35](https://cloud.githubusercontent.com/assets/1241240/15959660/16c1fee8-2efc-11e6-89cf-ac0f98e8e1f7.png)

Furthermore, starting from epoch 130, everything slows down a lot.

**Comment by aymericdamien** [comment]
ah? This is weird, I got good results for n = 9:
![selection_009](https://cloud.githubusercontent.com/assets/10386605/15959878/86488fc4-2f2f-11e6-9a22-16b7329993d9.png)

**Comment by aymericdamien** [comment]
Sure, you can go ahead and update the example with the results :)
I will investigate the issue with the `performance decrease, to see if I can reproduce it.

**Comment by daviddao** [comment]
What GPU are you using? It might also have been a caching issue. I was using a g2.2xlarge AWS instance (K520 Grid with 4 GB memory).

**Comment by aymericdamien** [comment]
I tested on Tesla k40c and GTX980
